### PR TITLE
Copter autotune messages

### PR DESCRIPTION
**Pull request by peterbarker**
Tell the user, at great length, what their vehicle is doing during autotune.

Some of these are cryptic, but some not - particularly "pilot override active"!

### CODE REVIEW

**Comment by peterbarker** [comment]
A 10-second interval of STATUSTEXT messages during autotune with this patch:

```
2017-05-13 15:12:55.09: MSG {TimeUS : 249841952, Message : PERF: 154/4002 3474 1984}
2017-05-13 15:12:56.21: MSG {TimeUS : 250959798, Message : AutoTune: Twitch}
2017-05-13 15:12:56.30: MSG {TimeUS : 251047818, Message : AutoTune: Rate D Down}
2017-05-13 15:12:56.30: MSG {TimeUS : 251048195, Message : AutoTune: TWITCHING}
2017-05-13 15:12:56.30: MSG {TimeUS : 251048455, Message : AutoTune: lean=290.0000 target=2000.000}
2017-05-13 15:12:56.30: MSG {TimeUS : 251048599, Message : AutoTune: rotation=73.59818 target=90.00000}
2017-05-13 15:12:56.30: MSG {TimeUS : 251048757, Message : AutoTune: rotation=73.59818 target=90.00000}
2017-05-13 15:12:56.30: MSG {TimeUS : 251048969, Message : AutoTune: p=0.037063 d=0.001714}
2017-05-13 15:12:56.30: MSG {TimeUS : 251049326, Message : AutoTune: success 0/4}
2017-05-13 15:12:57.39: MSG {TimeUS : 252142382, Message : AutoTune: Twitch}
2017-05-13 15:12:58.30: MSG {TimeUS : 253049125, Message : AutoTune: Rate D Down}
2017-05-13 15:12:58.30: MSG {TimeUS : 253049321, Message : AutoTune: WFL (Rate(R)) (10.09046 > 10.00000)}
2017-05-13 15:12:58.30: MSG {TimeUS : 253049468, Message : AutoTune: p=0.035210 d=0.001714}
2017-05-13 15:12:58.30: MSG {TimeUS : 253049656, Message : AutoTune: success 1/4}
2017-05-13 15:13:00.00: MSG {TimeUS : 254755560, Message : AutoTune: Twitch}
2017-05-13 15:13:00.30: MSG {TimeUS : 255050604, Message : AutoTune: Rate D Down}
2017-05-13 15:13:00.30: MSG {TimeUS : 255050842, Message : AutoTune: WFL (Angle(R)) (7.690000 > 3.000000)}
2017-05-13 15:13:00.30: MSG {TimeUS : 255051136, Message : AutoTune: p=0.035210 d=0.001629}
2017-05-13 15:13:00.30: MSG {TimeUS : 255051447, Message : AutoTune: success 0/4}
2017-05-13 15:13:00.94: MSG {TimeUS : 255692494, Message : AutoTune: Twitch}
2017-05-13 15:13:02.07: MSG {TimeUS : 256821502, Message : AutoTune: Twitch}
2017-05-13 15:13:02.30: MSG {TimeUS : 257053566, Message : AutoTune: Rate D Down}
2017-05-13 15:13:02.30: MSG {TimeUS : 257053680, Message : AutoTune: UPDATING_GAINS}
2017-05-13 15:13:02.30: MSG {TimeUS : 257053900, Message : AutoTune: lean=1471.000 target=2000.000}
2017-05-13 15:13:02.30: MSG {TimeUS : 257054278, Message : AutoTune: rotation=76.68346 target=90.00000}
2017-05-13 15:13:02.30: MSG {TimeUS : 257054533, Message : AutoTune: rotation=76.68346 target=90.00000}
2017-05-13 15:13:02.30: MSG {TimeUS : 257054684, Message : AutoTune: p=0.033449 d=0.001629}
2017-05-13 15:13:02.30: MSG {TimeUS : 257054876, Message : AutoTune: success 0/4}
2017-05-13 15:13:03.90: MSG {TimeUS : 258650287, Message : AutoTune: Twitch}
2017-05-13 15:13:04.30: MSG {TimeUS : 259055734, Message : AutoTune: Rate D Down}
2017-05-13 15:13:04.30: MSG {TimeUS : 259056034, Message : AutoTune: WFL (Angle(R)) (12.30000 > 3.000000)}
2017-05-13 15:13:04.31: MSG {TimeUS : 259056348, Message : AutoTune: p=0.033449 d=0.001629}
2017-05-13 15:13:04.31: MSG {TimeUS : 259056465, Message : AutoTune: success 1/4}
2017-05-13 15:13:05.26: MSG {TimeUS : 260008108, Message : PERF: 148/4001 4976 2210}
```

File: ArduCopter/control_autotune.cpp [hunk 1]
```diff
@@ -117,7 +119,7 @@ enum AutoTuneTuneType {
     AUTOTUNE_TYPE_RD_UP = 0,                // rate D is being tuned up
     AUTOTUNE_TYPE_RD_DOWN = 1,              // rate D is being tuned down
     AUTOTUNE_TYPE_RP_UP = 2,                // rate P is being tuned up
-    AUTOTUNE_TYPE_SP_DOWN = 3,              // angle P is being tuned up
+    AUTOTUNE_TYPE_SP_DOWN = 3,              // angle P is being tuned down
```
Comment (muramura) [inline ArduCopter/control_autotune.cpp]: SP does not match the official name, angle P. I think that it is better to change to AP or official name.

File: ArduCopter/control_autotune.cpp [hunk 2]
```diff
@@ -260,6 +273,126 @@ bool Copter::autotune_start(bool ignore_checks)
...
+    case Copter::AUTOTUNE_LEVEL_ISSUE_ANGLE_YAW:
+        return "Angle(Y)";
+    case Copter::AUTOTUNE_LEVEL_ISSUE_RATE_ROLL:
+        return "Rate(R)";
+    case Copter::AUTOTUNE_LEVEL_ISSUE_RATE_PITCH:
+        return "Rate(P)";
+    case Copter::AUTOTUNE_LEVEL_ISSUE_RATE_YAW:
+        return "Rate(Y)";
+    }
+    return "Bug";
```
Comment (muramura) [inline ArduCopter/control_autotune.cpp]: Multiple, "Bug" is defined. If the same pointer is OK. If it is different, I think that it is better to make it a common variable.

File: ArduCopter/control_autotune.cpp (hunk 1 above)
Comment (rmackay9) [inline ArduCopter/control_autotune.cpp]: @muramura, yes that's a good point.  That could be done as a separate PR though because it's an existing issue.

**Comment by rmackay9** [comment]
Merged, thanks.

**Pull request merged**